python/pyarrow/error.pxi

```diff
@@ -217,7 +218,10 @@ cdef class SignalStopHandler:
                 maybe_source.status().Warn()
             else:
                 self._stop_token.init(deref(maybe_source).token())
-                self._enabled = True
+                if not is_threading_enabled():
+                    self._enabled = False
```

Comment:
The signal handler (and so ctrl-c?) just doesn't work on emscripten, or this is actually threading related?

(potentially dumb question, I am not familiar with this part of our code base; this might be obvious, but it could maybe also use a clarifying code comment)

Reply:
I'm not sure of other platforms, but yes on Emscripten signal handling doesn't really work without another thread to fire off the signals and CTRL+C doesn't work.

Reply:
The tests for this handling all rely on threads also.